### PR TITLE
add logging, sync remove/read operations and handle exceptions

### PR DESCRIPTION
**Pull request by b2mdevelopment**
1) Add logging
2) Sync remove/adding and getting messages
3) Handle exceptions at Forward service to ensure that forward thread doesn't die when any exception occur.

### CODE REVIEW

**Comment by aflugge** [comment]
I abstain.

File: src/reyna/Store/SQLiteRepository.cs [hunk 1]
```diff
@@ -121,40 +121,49 @@ public void Add(IMessage message, long storageSizeLimit)
...
+                    var messageId = this.CreateParameter("@messageId", message.Id);
+                    this.ExecuteNonQuery(sql, t, messageId);
+                });
+            }
         }
 
         public IMessage GetNextMessageAfter(long messageId)
         {
-            return this.ExecuteInTransaction((t) =>
+            lock (Locker)
```
Comment (avovsya) [inline src/reyna/Store/SQLiteRepository.cs:145]: Why do you need to lock a transaction?

Comment (b2mdevelopment) [inline src/reyna/Store/SQLiteRepository.cs:145]: It raises database lock exception.
@Youhana-Hana

**Comment by kandeel** [comment]
reviewed

File: src/reyna/Store/SQLiteRepository.cs (hunk 1 above)
Comment (ejarlewski) [inline src/reyna/Store/SQLiteRepository.cs:145]: I guess because SQLite is not really meant for multithreaded environments?

**Comment by ejarlewski** [comment]
👍

**Pull request merged**